pkg/orchestrator/interfaces.go

```diff
@@ -95,21 +95,21 @@ type NodeDiscoverer interface {
 
 // NodeRanker ranks nodes based on their suitability to execute a job.
 type NodeRanker interface {
-	RankNodes(ctx context.Context, job models.Job, nodes []models.NodeInfo) ([]NodeRank, error)
+	RankNodes(ctx context.Context, job models.Job, retryDelay time.Duration, nodes []models.NodeInfo) ([]NodeRank, error)
 }
 
 // NodeSelector selects nodes based on their suitability to execute a job.
 type NodeSelector interface {
 	// AllNodes returns all nodes in the network.
 	AllNodes(ctx context.Context) ([]models.NodeInfo, error)
 	// AllMatchingNodes returns all nodes that match the job constrains and selection criteria.
-	AllMatchingNodes(ctx context.Context, job *models.Job) ([]models.NodeInfo, error)
+	AllMatchingNodes(ctx context.Context, job *models.Job, retryDelay time.Duration) ([]models.NodeInfo, error)
 	// TopMatchingNodes return the top ranked desiredCount number of nodes that match job constraints
 	// ordered in descending order based on their rank, or error if not enough nodes match.
-	TopMatchingNodes(ctx context.Context, job *models.Job, desiredCount int) ([]models.NodeInfo, error)
+	TopMatchingNodes(ctx context.Context, job *models.Job, retryDelay time.Duration, desiredCount int) ([]models.NodeInfo, error)
```

Comment:
if we follow my advice to encode the delay in the execution instead of the history, then we can pass the list of previous executions to the node rankers. We already have a ranker that pulls previous executions from the job store, and might make more sense to do it once and pass it to all rankers